system-variables: add configuration tip based on units (#6664)
pingcap/docs: system-variables.md

>
> TiDB differs from MySQL in that `GLOBAL` scoped variables **persist** through TiDB server restarts. Additionally, TiDB presents several MySQL variables as both readable and settable. This is required for compatibility, because it is common for both applications and connectors to read MySQL variables. For example, JDBC connectors both read and set query cache settings, despite not relying on the behavior.

> **Tip:**
>

> > Larger values will not always yield better performance. It is also important to consider the number of concurrent connections that are executing statements, as most settings apply per connection.

Reviewer: ```suggestion
> Larger values do not always yield better performance. It is also important to consider the number of concurrent connections that are executing statements, because most settings apply to each connection.
```

>

> > The 'unit' can be useful in determining safe values:

Reviewer: ```suggestion
> Consider the 'unit' of a variable when you determine safe values:
```

> > Consider the 'unit' of a variable when you determine safe values:

Reviewer: ```suggestion
> Consider the unit of a variable when you determine safe values:
```

>
> * For threads, safe values are typically up to the number of CPU cores.
> * For bytes, safe values are typically less than the amount of system memory.

> > * For time, pay attention that the unit could be Seconds or Milliseconds.

Reviewer: ```suggestion
> * For time, pay attention that the unit might be seconds or milliseconds.
```

> Variables using the same unit are likely to compete for the same set of resources.

Reviewer: ```suggestion
>
> Variables using the same unit might compete for the same set of resources.
```

>
> TiDB differs from MySQL in that `GLOBAL` scoped variables **persist** through TiDB server restarts. Additionally, TiDB presents several MySQL variables as both readable and settable. This is required for compatibility, because it is common for both applications and connectors to read MySQL variables. For example, JDBC connectors both read and set query cache settings, despite not relying on the behavior.

> **Note:**
>
> Larger values do not always yield better performance. It is also important to consider the number of concurrent connections that are executing statements, because most settings apply to each connection.
>
> The 'unit' can be useful in determining safe values:
>
> * For threads, safe values are typically up to the number of CPU cores.
> * For bytes, safe values are typically less than the amount of system memory.
> * For time, pay attention that the unit might be seconds or milliseconds.

> 

Reviewer: ```suggestion
```
